pkg/clustermesh/endpointslicesync/clustermesh.go

```diff
@@ -47,9 +47,34 @@ type clusterMesh struct {
 
 	endpointSliceMeshController  *endpointslice.Controller
 	endpointSliceInformerFactory informers.SharedInformerFactory
+
+	started                   bool
```

Comment:
Nit. I'd change this to an `atomic.Bool` to be on the safe side, and avoid possible race conditions.